docs/formats/zdm.md

```diff
@@ -0,0 +1,555 @@
+# The Zed Data Model
+
```

Comment:
Would be nice to have a table of contents so I can skip ahead.

Reply:
I was waiting till the end to add it.  It's a pain to keep up to date while editing.  Will add it before merging.